source/plugins/ruby/constants.rb

```diff
@@ -77,13 +77,17 @@ class Constants
   OMSAGENT_ZERO_FILL = "omsagent"
   KUBESYSTEM_NAMESPACE_ZERO_FILL = "kube-system"
   VOLUME_NAME_ZERO_FILL = "-"
+  PV_TYPES =["awsElasticBlockStore", "azureDisk", "azureFile", "cephfs", "cinder", "csi", "fc", "flexVolume",
+    "flocker", "gcePersistentDisk", "glusterfs", "hostPath", "iscsi", "local", "nfs",
+    "photonPersistentDisk", "portworxVolume", "quobyte", "rbd", "scaleIO", "storageos", "vsphereVolume"]
 
   #Telemetry constants
   CONTAINER_METRICS_HEART_BEAT_EVENT = "ContainerMetricsMdmHeartBeatEvent"
   POD_READY_PERCENTAGE_HEART_BEAT_EVENT = "PodReadyPercentageMdmHeartBeatEvent"
   CONTAINER_RESOURCE_UTIL_HEART_BEAT_EVENT = "ContainerResourceUtilMdmHeartBeatEvent"
   PV_USAGE_HEART_BEAT_EVENT = "PVUsageMdmHeartBeatEvent"
   PV_KUBE_SYSTEM_METRICS_ENABLED_EVENT = "CollectPVKubeSystemMetricsEnabled"
+  PV_INVENTORY_HEART_BEAT_EVENT = "KubePVInventoryHeartBeatEvent"
```

Comment:
Do you think we need hearbeat telemetry for this plugin ? We have for a few, but i dont think we use them much.

Reply:
The heartbeat itself is probably not needed. I am using it [here](https://github.com/microsoft/Docker-Provider/pull/455/files#diff-24c9dec7f951dc00ce99813aad1129dcR136) to collect telemetry on the PV types and the count of each type. This info might be nice to know for supporting more detailed type info for other types, even if it's sent with a very large interval. Do we want to collect this?

Reply:
ok. so you are using this event to get telemetry about pv types we see in the cluster as well. If so, please leave it as is. Yes, its very useful to have this telemetry.